libvast/vast/table_slice.hpp

```diff
@@ -203,6 +185,16 @@ class table_slice final {
              }));
   }
 
+  // -- operations -------------------------------------------------------------
+
+  /// Rebuilds a table slice with a given encoding if necessary.
+  /// @param slice The slice to rebuild.
+  /// @param encoding The encoding to convert to.
+  /// @note This function only rebuilds if necessary, i.e., the new encoding
+  /// is different from the existing one.
+  friend table_slice
+  rebuild_table_slice(table_slice slice, enum encoding encoding) noexcept;
```

Comment:
```suggestion
  rebuild(table_slice slice, enum encoding encoding) noexcept;
```

I think the arguments and documentation make it clear what's going on here.

Reply:
Fair enough, will change before merging tomorrow.